src/probnum/randprocs/_gauss_markov_process.py

```diff
@@ -0,0 +1,75 @@
+"""Gauss-Markov Processes."""
+from typing import Union
+
+import numpy as np
+
+from probnum import randvars, statespace
+from probnum.type import FloatArgType
+
+from ._gaussian_process import GaussianProcess
+
+_InputType = Union[np.floating, np.ndarray]
+_OutputType = Union[np.floating, np.ndarray]
+
+
+class GaussMarkovProcess(GaussianProcess):
+    r"""Gaussian processes with the Markov property.
+
+    A Gauss-Markov process is a Gaussian process with the additional property that
+    conditioned on the present state of the system its future and past states are
+    independent. This is known as the Markov property or as the process being
+    memoryless. Gauss-Markov processes :math:`x_t` can be defined as the solution of a
+    (linear) transition model given by a stochastic differential equation (SDE) of
+    the form
+
+    .. math:: \mathrm{d} x_t = G(t) x_t \mathrm{d} t + \mathrm{d} w_t.
+
+    and a Gaussian initial condition :math:`x_0`.
+
+    Parameters
+    ----------
+    linear_sde
+        Linear stochastic differential equation describing a state transition of the
+        system.
+    t0
+        Initial starting index / time of the process.
+    x0
+        Gaussian random variable describing the initial state.
+
+    See Also
+    --------
+    RandomProcess : Random processes.
+    GaussianProcess : Gaussian processes.
+
+    Examples
+    --------
+    """
+    # pylint: disable=invalid-name
+
+    def __init__(
+        self,
+        linear_sde: statespace.LinearSDE,
+        x0: randvars.Normal,
+        t0: FloatArgType = 0.0,
+    ):
+        self.linear_sde = linear_sde
+        self.t0 = t0
+        self.x0 = np.asarray(x0).reshape(1, -1)
+
+        super().__init__(
+            input_dim=1,
+            output_dim=1 if self.x0.ndim == 0 else self.x0.shape[0],
+            mean=lambda t: self.__call__(t).mean,
+            cov=lambda t: self.__call__(t).cov,
+        )
+
+    def __call__(self, x: _InputType) -> randvars.Normal:
+        """Closed form solution to the SDE evaluated at ``x`` as defined by the linear
+        transition."""
+        # TODO: currently the SDE Models do not support arbitrary steps as defined by
+        #  the increments in ``x``, only fixed step sizes ``dt```. This has to
+        #  be added first before __call__, mean, etc. work as defined by the
+        #  RandomProcess interface.
+
+        # return self.linear_sde.forward_rv(rv=self.x0, t=self.t0, dt=x[-1] - x[-2])
```

Comment:
At first sight (and sorry for the directness, it is not meant badly!) I dont think this is true.

the code `self.linear_sde.forward_rv(rv=self.x0, t=self.t0, dt=(t-self.t0)` should work in order to evaluate the GM process at a single point `t`. Evaluating it at multiple steps `(t1, t2, ...)` is a different story. It is possible (I can post a link to how), but neither is it readily available (I've never needed it), nor is it trivial to implement.

Reply:
Maybe this would also motivate providing the interface for at least a `cov` function (the mean function should be possible directly via forward_rv things). This could then properly subclass the GP interface (because now there would be a real cov), and would help encapsulating a rather weird function.

Reply:
Maybe I misunderstand, so please correct me if I am wrong. If I want to evaluate a Gauss-Markov process at time points `(t_0, t_1, ..., t_n)` not necessarily equidistant, then from looking at the interface of `forward_rv` I cannot do this in a single call. However I guess I could just iteratively apply it with new starting point `t_i` and `initrv` the output of forward_rv? The point is that the random process interface lets you evaluate at multiple inputs in a single call. For this PR I would leave the MarkovProcess simply as an interface if that is okay with you. You are much more familiar with your code than I am.